deepchem/data/datasets.py

```diff
@@ -2465,29 +2638,43 @@ def iterate(dataset, batch_size, epochs, deterministic, pad_batches):
 
     return iterate(self, batch_size, epochs, deterministic, pad_batches)
 
-  def itersamples(self) -> Iterator[Batch]:
-    """Get an object that iterates over the samples in the dataset.
+  def _get_image(self, array: Union[np.ndarray, List[str]],
+                 index: int) -> np.ndarray:
+    """Method for loading an image
 
-    Example:
+    Parameters
+    ----------
+    array: Union[np.ndarray, List[str]]
+      A numpy array which contains images or List of image filenames
+    index: int
+      Index you want to get the image
 
-    >>> dataset = NumpyDataset(np.ones((2,2)))
-    >>> for x, y, w, id in dataset.itersamples():
-    ...   print(x.tolist(), y.tolist(), w.tolist(), id)
-    [1.0, 1.0] [0.0] [0.0] 0
-    [1.0, 1.0] [0.0] [0.0] 1
+    Returns
+    -------
+    np.ndarray
+      Loaded image
     """
+    if isinstance(array, np.ndarray):
+      return array[index]
+    return load_image_files([array[index]])[0]
 
-    def get_image(array, index):
-      if isinstance(array, np.ndarray):
-        return array[index]
-      return load_image_files([array[index]])[0]
+  def itersamples(self) -> Iterator[Batch]:
+    """Get an object that iterates over the samples in the dataset.
 
+    Returns
+    -------
+    Iterator[Batch]
+      Iterator which yields tuples of four numpy arrays `(X, y, w, ids)`.
+    """
     n_samples = self._X_shape[0]
-    return ((get_image(self._X, i), get_image(self._y, i), self._w[i],
-             self._ids[i]) for i in range(n_samples))
-
-  def transform(self, transformer: "dc.trans.Transformer",
-                **args) -> NumpyDataset:
+    return ((self._get_image(self._X, i), self._get_image(self._y, i),
+             self._w[i], self._ids[i]) for i in range(n_samples))
+
+  def transform(
+      self,
+      transformer: "dc.trans.Transformer",
+      **args,
+  ) -> "ImageDataset":
```

Comment:
The transformed dataset is a NumpyDataset, not an ImageDataset.

Reply:
OK. I revert this modification